### PR TITLE
Fix warnings during the "Bundling to UMD"

### LINKED ISSUE #49: Fix WARNING: No name was provided for external module "xxx" in output.globals – guessing "xxx"
**Issue in terminal when building the library knora-ui**
Screenshot 2020-03-19 at 10 02 11

**Solution**
- add the following lines to `ng-package.json` and `ng-package.prod.json`
```
{
    ...
    "lib": {
       ...,
        "umdModuleIds": {
            "@knora/api": "@knora/api",
            "ngx-color-picker": "ngxColorPicker",
            "jdnconvertiblecalendar": "jdnconvertiblecalendar",
            "jdnconvertiblecalendardateadapter": "jdnconvertiblecalendardateadapter",
            "jdnconvertiblecalendar/dist/src/JDNConvertibleCalendar": "JDNConvertibleCalendar"
        }
    }
}
```

### PR DESCRIPTION
**Pull request by flavens**
It resolves the following warnings during the project build:
Screenshot 2020-03-26 at 09 43 08

closes #49 

YouTrack issue: https://dasch.myjetbrains.com/youtrack/issue/DSP-51

### CODE REVIEW

File: projects/knora-ui/ng-package.json [hunk 1]
```diff
@@ -2,6 +2,13 @@
...
   "dest": "../../dist/knora-ui",
   "lib": {
-    "entryFile": "src/public-api.ts"
+    "entryFile": "src/public-api.ts",
+    "umdModuleIds": {
+      "@knora/api": "@knora/api",
+      "ngx-color-picker": "ngxColorPicker",
+      "jdnconvertiblecalendar": "jdnconvertiblecalendar",
+      "jdnconvertiblecalendardateadapter": "jdnconvertiblecalendardateadapter",
+      "jdnconvertiblecalendar/dist/src/JDNConvertibleCalendar": "JDNConvertibleCalendar"
```
Comment (tobiasschweizer) [inline projects/knora-ui/ng-package.json]: I wonder why there is `jdnconvertiblecalendar` and `JDNConvertibleCalendar`.  `jdnconvertiblecalendar/dist/src/JDNConvertibleCalendar` seems odd. Maybe I did something wrong in https://github.com/dhlab-basel/JDNConvertibleCalendar, I will check this.

**Comment by tobiasschweizer** [comment]
I found the problem:

https://github.com/dasch-swiss/knora-ui-ng-lib/blob/373e762ac7dcab0f8c9894fde830c749aba96ca7/projects/knora-ui/src/lib/viewer/values/time-value/time-input/time-input.component.ts#L7-L9

The imports have to be changed to (`JDNConvertibleCalendarModule` is not needed): 

```ts
import {CalendarDate, CalendarPeriod, GregorianCalendarDate} from 'jdnconvertiblecalendar';
```

and then 

https://github.com/dasch-swiss/knora-ui-ng-lib/blob/6d211c58da73c7c4eb8695a0c6c68fc2fca1bb62/projects/knora-ui/ng-package.json#L11

becomes obsolete.

**Comment by tobiasschweizer** [comment]
I wonder if we have to do something about

> WARNING: 'ReadValue', 'ReadTextValueAsString', 'ReadIntValue', 'ReadLinkValue', 'ReadBooleanValue', 'ReadDecimalValue', 'ReadColorValue', 'ReadUriValue', 'ReadIntervalValue' and 'ReadTimeValue' are imported from external module '@knora/api' but never used

I found https://github.com/angular/angular/issues/21280 which you have already mentioned :-)

**Comment by flavens** [comment]
> I found angular/angular#21280 which you have already mentioned :-)

I tried the solution with `emitDecoratorMetadata` sets to true but it did not change anything...

**Comment by tobiasschweizer** [comment]
> > I found [angular/angular#21280](https://github.com/angular/angular/issues/21280) which you have already mentioned :-)
> 
> I tried the solution with `emitDecoratorMetadata` sets to true but it did not change anything...

I think it's not a real problem!

**Comment by tobiasschweizer** [comment]
probably it's just some imports that are still in the JavaScript code that's generated from the TypeScript source, but in JavaScript these classes are not used since the refer to TypeScript constructs

**Comment by tobiasschweizer** [comment]
so this PR is good to go :-)

**Pull request merged**